cloud/types.go

```diff
@@ -219,14 +234,45 @@ type (
 		GithubActionsWorkflowRef           string `json:"github_actions_workflow_ref,omitempty"`
 	}
 
-	// GitMetadata are the git related metadata.
-	GitMetadata struct {
-		GitCommitSHA         string     `json:"git_commit_sha,omitempty"`
-		GitCommitAuthorName  string     `json:"git_commit_author_name,omitempty"`
-		GitCommitAuthorEmail string     `json:"git_commit_author_email,omitempty"`
-		GitCommitAuthorTime  *time.Time `json:"git_commit_author_time,omitempty"`
-		GitCommitTitle       string     `json:"git_commit_title,omitempty"`
-		GitCommitDescription string     `json:"git_commit_description,omitempty"`
+	// GitlabMetadata holds the Gitlab specific metadata.
+	GitlabMetadata struct {
+		GitlabMergeRequestAuthorID        int    `json:"gitlab_merge_request_author_id,omitempty"`
+		GitlabMergeRequestAuthorName      string `json:"gitlab_merge_request_author_name,omitempty"`
+		GitlabMergeRequestAuthorWebURL    string `json:"gitlab_merge_request_author_web_url,omitempty"`
+		GitlabMergeRequestAuthorUsername  string `json:"gitlab_merge_request_author_username,omitempty"`
+		GitlabMergeRequestAuthorAvatarURL string `json:"gitlab_merge_request_author_avatar_url,omitempty"`
+		GitlabMergeRequestAuthorState     string `json:"gitlab_merge_request_author_state,omitempty"`
+
+		GitlabMergeRequestID           int    `json:"gitlab_merge_request_id,omitempty"`
+		GitlabMergeRequestIID          int    `json:"gitlab_merge_request_iid,omitempty"`
+		GitlabMergeRequestState        string `json:"gitlab_merge_request_state,omitempty"`
+		GitlabMergeRequestCreatedAt    string `json:"gitlab_merge_request_created_at,omitempty"`
+		GitlabMergeRequestUpdatedAt    string `json:"gitlab_merge_request_updated_at,omitempty"`
+		GitlabMergeRequestTargetBranch string `json:"gitlab_merge_request_target_branch,omitempty"`
+		GitlabMergeRequestSourceBranch string `json:"gitlab_merge_request_source_branch,omitempty"`
+		GitlabMergeRequestMergeStatus  string `json:"gitlab_merge_request_merge_status,omitempty"`
+		GitlabMergeRequestWebURL       string `json:"gitlab_merge_request_web_url,omitempty"`
+
+		// CICD
+		GitlabCICDJobManual         bool   `json:"gitlab_cicd_job_manual,omitempty"`          // CI_JOB_MANUAL
+		GitlabCICDPipelineID        string `json:"gitlab_cicd_pipeline_id,omitempty"`         // CI_PIPELINE_ID
+		GitlabCICDPipelineSource    string `json:"gitlab_cicd_pipeline_source,omitempty"`     // CI_PIPELINE_SOURCE
+		GitlabCICDPipelineTriggered bool   `json:"gitlan_cicd_pipeline_triggered,omitempty"`  // CI_PIPELINE_TRIGGERED
+		GitlabCICDPipelineURL       string `json:"gitlab_cicd_pipeline_url,omitempty"`        // CI_PIPELINE_URL
```

Comment:
The CI_PIPELINE_NAME is missing here, could you add it as part of `gitlab_cicd_pipeline_name`?

Reply:
It's added now. Thanks